index.js

```diff
@@ -22,7 +22,22 @@ function onexit () {
   exiting = true
 
   process.removeListener('beforeExit', onexit)
-  Promise.allSettled(handlers.map(fn => fn())).then(done, done)
+
+  let promise = null
+  const positions = [...map.keys()].sort()
```

Comment:
pass a compare function to sort, otherwise it's lexicographic

(a, b) => a - b